connector/servicegraphconnector/connector.go

```diff
@@ -606,12 +606,12 @@ func (p *serviceGraphConnector) buildMetricKey(clientName, serverName, connectio
 	var metricKey strings.Builder
 	metricKey.WriteString(clientName + metricKeySeparator + serverName + metricKeySeparator + connectionType + metricKeySeparator + failed)
 
-	for _, dimName := range p.config.Dimensions {
-		dim, ok := edgeDimensions[dimName]
-		if !ok {
-			continue
+	for _, kind := range []string{serverKind, clientKind} {
+		for _, dimName := range p.config.Dimensions {
+			if dim, ok := edgeDimensions[kind+"_"+dimName]; ok {
```

Comment:
```suggestion
			if dim, ok := edgeDimensions[kind + "_" + dimName]; ok {
```
Nit: Whitespace around `+` operator.

Reply:
There is no space in the original code, such as in the `upsertDimensions` function. I think it is better to keep the original style.

Reply:
My IDE `GoLand` also removes the spaces on both sides of the `+` operate when I save.